lib/pure/md5.nim

```diff
@@ -237,29 +271,41 @@ proc md5Init*(c: var MD5Context) =
   c.state[3] = 0x10325476'u32
   c.count[0] = 0'u32
   c.count[1] = 0'u32
-  zeroMem(addr(c.buffer), sizeof(MD5Buffer))
+  when false:
+    zeroMem(addr(c.buffer), sizeof(MD5Buffer))
+  else:
+    reset(c.buffer)
 
-proc md5Update*(c: var MD5Context, input: cstring, len: int) =
-  ## Updates the `MD5Context` with the `input` data of length `len`.
+proc md5Update*(c: var MD5Context, input: openArray[uint8]) =
+  ## Updates the `MD5Context` with the `input` data.
   ##
   ## If you use the `toMD5 proc <#toMD5,string>`_, there's no need to call this
   ## function explicitly.
-  var input = input
+  #var input = input
   var Index = int((c.count[0] shr 3) and 0x3F)
-  c.count[0] = c.count[0] + (uint32(len) shl 3)
-  if c.count[0] < (uint32(len) shl 3): c.count[1] = c.count[1] + 1'u32
-  c.count[1] = c.count[1] + (uint32(len) shr 29)
+  c.count[0] = c.count[0] + (uint32(input.len) shl 3)
+  if c.count[0] < (uint32(input.len) shl 3): c.count[1] = c.count[1] + 1'u32
+  c.count[1] = c.count[1] + (uint32(input.len) shr 29)
   var PartLen = 64 - Index
-  if len >= PartLen:
-    copyMem(addr(c.buffer[Index]), input, PartLen)
-    transform(addr(c.buffer), c.state)
+  if input.len >= PartLen:
+    when false:
+      copyMem(addr(c.buffer[Index]), unsafeAddr(input[0]), PartLen)
+    else:
+      c.buffer[Index .. Index + PartLen - 1] = input.slice(0, PartLen - 1)
```

Comment:
`[]=` with openarray is pure Nim and [raises RangeDefect](https://github.com/nim-lang/Nim/blob/afa4bc34b41307c6c63e8c7c973afe295e9fedd8/lib/system.nim#L2607) without a compile-time precondition to raise it which means the `raises: []` claim above fails if defects are tracked, as in the CI error, even though this should never raise `RangeDefect` . So the choice is between using `{.cast(raises: []).}` here (or inside `[]=`) or saying `raises: [RangeDefect]`/`raises: [Defect]` above.